tests/UnitTests/TelemetryKeyCacheTests.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.ApplicationInsights.Kubernetes
+{
+    public class TelemetryKeyCacheTests
+    {
+        [Theory]
+        [InlineData("abc.def", "abc_def")]
+        [InlineData("aaa.bbb.ccc.ddd", "aaa_bbb_ccc_ddd")]
+        public void ShouldAlterKeyNameWhenSet(string input, string expected)
+        {
+            AppInsightsForKubernetesOptions opts = new AppInsightsForKubernetesOptions();
+            opts.TelemetryKeyProcessor = (key) => key.Replace('.', '_');
+
+            IOptions<AppInsightsForKubernetesOptions> options = Options.Create<AppInsightsForKubernetesOptions>(opts);
+            TelemetryKeyCache keyCache = new TelemetryKeyCache(options);
+
+            string actual = keyCache.GetTelemetryProcessedKey(input);
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void AllowsMoreThanJustReplaceDot()
+        {
+            string input = "abc.def";
+            string expected = "\"abc.def\"";
+
+            AppInsightsForKubernetesOptions opts = new AppInsightsForKubernetesOptions();
+            opts.TelemetryKeyProcessor = (key) => $@"""{key}"""; // Add quotes
+
+            IOptions<AppInsightsForKubernetesOptions> options = Options.Create<AppInsightsForKubernetesOptions>(opts);
+            TelemetryKeyCache keyCache = new TelemetryKeyCache(options);
+
+            string actual = keyCache.GetTelemetryProcessedKey(input);
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ReturnsOriginalWhenProcessorNotSet()
```

Comment:
Could you add a test that succeeds on parsing "Abc.Def" to "AbcDef"?

Reply:
That would be easy to add. Mind explain me why?

Reply:
I just think that some team's conventions could be replacing Kubernetes.Container.Name for KubernetesContainerName, for example. So just want to make sure the test for that scenario passes to assure it is a possibility.